packages/hs-react-ui/src/components/Rating/Rating.tsx

```diff
@@ -0,0 +1,413 @@
+      if (!isRatingSelectedRef.current) {
+        setRatingValue(processRating(((event.clientX - rect.left) / rect.width) * max));
+      }
+    }
+  };
+  // get everything we expose + anything consumer wants to send to container
+  const mergedContainerProps = {
+    id,
+    elevation,
+    color: containerColor,
+    variant,
+    ...containerProps,
+  };
+
+  const clickHandler = (e: React.MouseEvent) => {
+    e.preventDefault();
+    setPreHoverRatingValue(ratingValue);
+    setRatingSelected(!isRatingSelected);
+    onClick(ratingValue);
+  };
+
+  const mouseLeaveHandler = (e: React.MouseEvent) => {
+    e.preventDefault();
+    setRatingValue(preHoverRatingValue);
+  };
+
+  useEffect(() => {
+    const current: HTMLElement | null = containerRef
+      ? ((containerRef.current as unknown) as HTMLElement)
+      : null;
+    if (current) {
+      current.addEventListener('mousemove', ratingListener);
+    }
+
+    if (value !== initialValue && !disabled) {
+      setRatingValue(processRating(value));
+      setPreHoverRatingValue(processRating(value));
+      setInitialValue(processRating(value));
+    }
+
+    if (disabled) {
+      setPreDisableRatingSelected(isRatingSelected);
+      setRatingSelected(true);
+    } else {
+      setRatingSelected(preDisableRatingSelected);
+    }
+    return () => {};
+  }, [value, fractionStep, max, showDisplay, disabled]);
+
+  const ratings: Array<JSX.Element> = [];
+
+  if (stages) {
+    for (let x = stages.length; x > 0; x--) {
+      if (x <= Math.floor(ratingValue)) {
+        ratings.push(
+          <StyledFilledRank onClick={clickHandler} id={x} key={x} {...filledRankProps}>
+            {typeof stages[x] === 'string' && stages[x - 1] !== '' ? (
+              <UnstyledIcon path={stages[x - 1] as string} size="2rem" />
+            ) : (
+              stages[x - 1]
+            )}
+          </StyledFilledRank>,
+        );
+      } else {
+        ratings.push(
+          <StyledEmptyRank
+            hasStages={hasStages}
+            ratingSelected={isRatingSelected}
+            onClick={clickHandler}
+            id={x}
+            key={x}
+            {...emptyRankProps}
+          >
+            {typeof stages[x] === 'string' && stages[x - 1] !== '' ? (
+              <UnstyledIcon path={stages[x - 1] as string} size="2rem" />
+            ) : (
+              stages[x - 1]
+            )}
+          </StyledEmptyRank>,
+        );
+      }
+    }
+  } else {
+    // No rating stages. Use filled, halfFilled, and empty ratings
+    for (let x = max; x > 0; x--) {
+      if (x <= Math.round(ratingValue - 0.01)) {
+        ratings.push(
+          <StyledFilledRank onClick={clickHandler} id={x} key={x} {...filledRankProps}>
+            {typeof filledRankItem === 'string' && filledRankItem !== '' ? (
+              <UnstyledIcon path={filledRankItem} size="2rem" />
+            ) : (
+              filledRankItem
+            )}
+          </StyledFilledRank>,
```

Comment:
StyledFilledRank and filledRankItem are confusing to me and StyledFilledRank looks more like a container than rank. Is that accurate? Same for half and empty styled ranks.

Reply:
- [ ] Rename containers